ibm/service/power/resource_ibm_pi_instance_action.go

```diff
@@ -0,0 +1,213 @@
+				Description:  "PVM instance action type",
+			},
+			Arg_PVMInstanceHealthStatus: {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validate.ValidateAllowedStringValues([]string{PVMInstanceHealthOk, PVMInstanceHealthWarning}),
+				Default:      "OK",
+				Description:  "Set the health status of the PVM instance to connect it faster",
+			},
+
+			// Computed
+			Attr_Status: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the PVM instance",
+			},
+			Attr_Progress: {
+				Type:        schema.TypeFloat,
+				Computed:    true,
+				Description: "The progress of an operation",
+			},
+			Attr_HealthStatus: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The PVM's health status value",
+			},
+		},
+	}
+}
+
+func resourceIBMPIInstanceActionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	adiag := takeInstanceAction(ctx, d, meta, d.Timeout(schema.TimeoutCreate))
+	if adiag != nil {
+		return adiag
+	}
+
+	return resourceIBMPIInstanceActionRead(ctx, d, meta)
+}
+
+func resourceIBMPIInstanceActionRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	id := d.Get(Arg_PVMInstanceId).(string)
+
+	client := st.NewIBMPIInstanceClient(context.Background(), sess, cloudInstanceID)
+	powervmdata, err := client.Get(id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.Set(Attr_Status, powervmdata.Status)
+	d.Set(Attr_Progress, powervmdata.Progress)
+	if powervmdata.Health != nil {
+		d.Set(Attr_HealthStatus, powervmdata.Health.Status)
+	}
+
+	d.SetId(fmt.Sprintf("%s/%s", cloudInstanceID, *powervmdata.PvmInstanceID))
+
+	return nil
+}
+
+func resourceIBMPIInstanceActionUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+
+	if d.HasChange(Arg_PVMInstanceActionType) {
+		takeInstanceAction(ctx, d, meta, d.Timeout(schema.TimeoutUpdate))
+	}
+
+	return resourceIBMPIInstanceActionRead(ctx, d, meta)
+}
+
+func resourceIBMPIInstanceActionDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// There is no delete or unset concept for instance action
+	d.SetId("")
+	return nil
+}
+
+func takeInstanceAction(ctx context.Context, d *schema.ResourceData, meta interface{}, timeout time.Duration) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	id := d.Get(Arg_PVMInstanceId).(string)
+	action := d.Get(Arg_PVMInstanceActionType).(string)
+	targetHealthStatus := d.Get(Arg_PVMInstanceHealthStatus).(string)
+
+	body := &models.PVMInstanceAction{Action: flex.PtrToString(action)}
```

Comment:
What is the benefit to calling PtrToString instead of just using "&" ?

Reply:
yeah no benefit here, removing it.